### DISCUSSION TITLE
Access collection tree via id?

### DISCUSSION THREAD

**Discussion by stephenmeehanuk** [statamic/cms#6101] [General]
Hi, 

I'm using `products_entries_field` to cherry pick `entires` from `collection:products`. `collection:products` is being used as a navigation. 

In  `collection:divisions` l'd like to pick a page from from `collection:products` and display any child pages.

I've got this so far.

```
{{ products_entries_field }}
    {{ nav:collection:products as="entries" id:contains="{id}"}}
        <ul> 
            {{ entries }}              
                <li><a href="{{ url }}">{{ title }}</a></li>
                    {{ if children }}              
                        {{ children }}
                            <li><a href="{{ url }}">{{ title }}</a></li>
                        {{ /children }}
                    {{ /if }}
            {{ /entries }}
        </ul>                
    {{ /nav:collection:products }}
{{ /products_entries_field }}
```

This will show the parent, but not the children?

**Comment by edalzell** [comment]
Products is an ordered collection?

**Reply by stephenmeehanuk** [reply]
Yes, `products_entries_field` is in `collection:divisions`, I'm using it to associate entries from `collection:products`

`collection:products` is an ordered collection, I'm using two blueprints to create parent and child pages.

On `divisions/show` I can display `products_entries_field` like this:

```
{{ products_entries_field as="items"}}
    <ul>
        {{ items }}
            <li><a href="{{ url }}">{{ title }}</a></li>
        {{ /items }}
    </ul>
{{ /products_entries_field }
```
Some of the entires from `products_entries_field` have child pages. 

I'm was hoping to be able to display them within that loop.

Essentially, I'm trying to do this, but this doesn't work...
```
{{ products_entries_field as="items"}}
    <ul>
        {{ items }}
            <li><a href="{{ url }}">{{ title }}</a>
                {{ if children }}
                    <ul>
                        {{ children }}
                        <li><a href="{{ url }}">{{ title }}</a></li>
                        {{ /children }}
                    </ul>  
                {{ /if }}
            </li>   
        {{ /items }}
    </ul>
{{ /products_entries_field }}
```

**Comment by JohnathonKoster** [comment]
If the `products` collection has routes, the following is one way that it could be done (someone else may know a cleaner way):

```antlers
<ul>
{{ products_entries_field }}
    <li>
    {{ title }}

    {{#
            Get the nav structure starting at the current product.
            Note: I chose 5 for max_depth arbitrarily
     #}}
    {{ nav:collection:products :from="url" as="entries" max_depth="5" }}
        {{#
            The start of our inner loop.
            This will only show the inner ul if there are any items.
        #}}
        {{ if entries | count > 0 }}
        <ul>
            {{# Loop through the entries like normal. #}}
            {{ entries }}
            <li>{{ title }}</li>

            {{#
                We can now do our children array (with recursion if we want).
            #}}
            {{ if children }}
                <ul>
                    {{ children }}
                        <li>
                            {{ title }}
                        
                            {{ if children }}
                                <ul>{{ *recursive children* }}</ul>
                            {{ /if }}
                        </li>
                    {{ /children }}
                </ul>
            {{ /if }}
            {{ /entries }}
        </ul>
        {{ /if }}
    {{ /nav:collection:products }}
    </li>
{{ /products_entries_field }}
</ul>
```

I set up a local test like so:

<img width="902" alt="nav_setup" src="https://user-images.githubusercontent.com/5232890/170841808-f44f5d0e-1169-4011-81c7-f1e090469daa.png">

And the above template produces the following result:

<img width="545" alt="final_result" src="https://user-images.githubusercontent.com/5232890/170841822-c8ca67aa-ff34-4428-adf6-d9990b5146e9.png">

I think in the original example, the `id:contains="{id}"` part of the query is being applied to all of the children, causing them to be excluded.

**Reply by stephenmeehanuk** [reply]
That's great, thanks so much for taking a look at this. It works!

I added `{{ if products_entries_field }} ... {{ /if }}` around the entire loop, as the page threw an error if `products_entries_field` didn't exist.
